src/applications/personalization/profile360/containers/PaymentInformation.jsx

```diff
@@ -144,103 +144,86 @@ class PaymentInformation extends React.Component {
-    );
-  }
-
   render() {
-    if (!this.props.isEligible) {
-      return null;
-    }
-
-    if (this.props.isLoading) {
-      return <LoadingIndicator message="Loading payment information..." />;
-    }
-
-    const { paymentInformation } = this.props;
+    const {
+      isEligible,
+      isLoading,
+      multifactorEnabled,
+      paymentInformation,
+    } = this.props;
     const directDepositIsSetUp =
       paymentInformation &&
       get('responses[0].paymentAccount.accountNumber', paymentInformation);
 
     let content = null;
 
-    if (!this.props.multifactorEnabled) {
+    if (!isEligible) {
+      return content;
+    }
+
+    if (isLoading) {
+      return <LoadingIndicator message="Loading payment information..." />;
+    }
+
+    if (!multifactorEnabled) {
       content = <PaymentInformation2FARequired />;
     } else if (paymentInformation.error) {
       content = <LoadFail information="payment" />;
+    } else if (!directDepositIsSetUp) {
+      // we might eventually want to return some helpful contenct explaining how
+      // a vet could go about setting up direct deposit
+      return null;
     } else {
-      const paymentAccount = paymentInformation.responses[0].paymentAccount;
-
-      content = (
-        <>
-          <div className="vet360-profile-field">
-            <ProfileFieldHeading
-              onEditClick={
-                directDepositIsSetUp &&
-                (() => this.handleLinkClick('edit', 'bank-name'))
-              }
-            >
-              Bank name
-            </ProfileFieldHeading>
-            {directDepositIsSetUp
-              ? paymentAccount.financialInstitutionName
-              : this.renderSetupButton('bank name', 'bank-name')}
-          </div>
-          <div className="vet360-profile-field">
-            <ProfileFieldHeading
-              onEditClick={
-                directDepositIsSetUp &&
-                (() => this.handleLinkClick('edit', 'account-number'))
-              }
-            >
-              Account number
-            </ProfileFieldHeading>
-            {directDepositIsSetUp
-              ? paymentAccount.accountNumber
-              : this.renderSetupButton('account number', 'account-number')}
-          </div>
-          <div className="vet360-profile-field">
-            <ProfileFieldHeading
-              onEditClick={
-                directDepositIsSetUp &&
-                (() => this.handleLinkClick('edit', 'account-type'))
-              }
-            >
-              Account type
-            </ProfileFieldHeading>
-            {directDepositIsSetUp
-              ? paymentAccount.accountType
-              : this.renderSetupButton(
-                  'account type (checking or savings)',
-                  'account-type',
-                )}
-          </div>
-          {directDepositIsSetUp && (
+      const paymentAccount = paymentInformation?.responses[0]?.paymentAccount;
+      if (paymentAccount) {
```

Comment:
adding this check adds a level of safety, but always introduced a little bit of extra noise to this diff; some of the below additions are really just due to indentation changing.

Reply:
Oh good catch. This additional check isn't actually needed